src/test/java/bio/terra/common/retry/SamRetryTest.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.common.retry;
+
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.client.http.HttpStatusCodes;
+import java.time.Duration;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("unit")
+public class SamRetryTest {
+
+  private int count;
+
+  @BeforeEach
+  public void setup() {
+    count = 0;
+  }
+
+  @Test
+  public void testRetryTimeout() throws Exception {
+    assertThrows(
+        ApiException.class,
+        () -> SamRetry.retry(() -> testRetryFinishInner(100), Duration.ofSeconds(10)));
```

Comment:
Let's try to add one test where we throw an InterruptedException by starting the work on another thread and interrupting it while it's sleeping on its retry.

Reply:
+1

Reply:
Added testInterruptedExceptionThrown below